core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -280,6 +280,8 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
 
   private[spark] def ui: Option[SparkUI] = _ui
 
+  def uiWebUrl: Option[String] = _ui.map(_.webUrl)
```

Comment:
PS on second thought maybe we can make this new method private[spark]. Or, how about not adding it? it really doesn't add much in return for adding a whole new API method.